auth-backends/src/main/scala/bootstrap/rudder/plugin/AuthBackendsConf.scala

```diff
@@ -484,13 +499,68 @@ trait RudderUserServerMapping[R <: OAuth2UserRequest, U <: OAuth2User, T <: Rudd
 
 }
 
-class RudderOidcUserService(rudderUserDetailsService: RudderInMemoryUserDetailsService)
-    extends OidcUserService with RudderUserServerMapping[OidcUserRequest, OidcUser, RudderUserDetail with OidcUser] {
+class RudderOidcUserService(
+    rudderUserDetailsService: RudderInMemoryUserDetailsService,
+    registrationRepository:   RudderClientRegistrationRepository
+) extends OidcUserService with RudderUserServerMapping[OidcUserRequest, OidcUser, RudderUserDetail with OidcUser] {
   // we need to use our copy of DefaultOAuth2UserService to log/manage errors
   super.setOauth2UserService(new RudderDefaultOAuth2UserService())
 
   override def loadUser(userRequest: OidcUserRequest): OidcUser = {
-    mapRudderUser("OIDC", super.loadUser(_), rudderUserDetailsService, userRequest, new RudderOidcDetails(_, _))
+    def buildUser(oidc: OidcUser, rudder: RudderUserDetail): RudderOidcDetails = {
+      val roles = {
+        registrationRepository.registrations.get(userRequest.getClientRegistration.getRegistrationId) match {
+          case None      =>
+            AuthBackendsLogger.trace(
+              s"No configuration found for OIDC registration id: ${userRequest.getClientRegistration.getRegistrationId}"
+            )
+            rudder.roles // if no registration, use user roles
+          case Some(reg) =>
+            if (reg.roles.enabled) {
+              val custom = {
+                try {
```

Comment:
here is the main part that hav impact on role, ie the security aspect. 
- we use standard oidc data type, 
- findRoleByName only looks for existing role with the same name (case insensitive, like other role resolution in rudder)